controllers/reconcilers/kserve_istio_smm_reconciler.go

```diff
@@ -0,0 +1,147 @@
+	}
+
+	// Get Existing resource
+	existingResource, err := r.getExistingResource(ctx, log, isvc.Namespace)
+	if err != nil {
+		return err
+	}
+
+	// Process Delta
+	if err = r.processDelta(ctx, log, desiredResource, existingResource); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *KserveServiceMeshMemberReconciler) Cleanup(ctx context.Context, log logr.Logger, namespace string) error {
+	existingSMM, getError := r.getExistingResource(ctx, log, namespace)
+	if getError != nil {
+		return getError
+	}
+
+	return r.processDelta(ctx, log, nil, existingSMM)
+}
+
+func (r *KserveServiceMeshMemberReconciler) createDesiredResource(ctx context.Context, isvc *kservev1beta1.InferenceService) (*v1.ServiceMeshMember, error) {
+	smcpName, smcpNamespace := utils.GetIstioControlPlaneName(ctx, r.client)
+
+	return &v1.ServiceMeshMember{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      constants.ServiceMeshMemberName,
+			Namespace: isvc.Namespace,
+			Labels: map[string]string{
+				"app.kubernetes.io/name":       "odh-model-controller",
+				"app.kubernetes.io/component":  "kserve",
+				"app.kubernetes.io/part-of":    "odh-model-serving",
+				"app.kubernetes.io/managed-by": "odh-model-controller",
+			},
+			Annotations: nil,
+		},
+		Spec: v1.ServiceMeshMemberSpec{
+			ControlPlaneRef: v1.ServiceMeshControlPlaneRef{
+				Name:      smcpName,
+				Namespace: smcpNamespace,
+			},
+		},
+	}, nil
+}
+
+func (r *KserveServiceMeshMemberReconciler) getExistingResource(ctx context.Context, log logr.Logger, namespace string) (*v1.ServiceMeshMember, error) {
+	return r.smmHandler.Fetch(ctx, log, namespace, constants.ServiceMeshMemberName)
+}
+
+func (r *KserveServiceMeshMemberReconciler) processDelta(ctx context.Context, log logr.Logger, desiredSMM *v1.ServiceMeshMember, existingSMM *v1.ServiceMeshMember) error {
+	comparator := comparators.GetServiceMeshMemberComparator()
+	delta := r.deltaProcessor.ComputeDelta(comparator, desiredSMM, existingSMM)
+
+	if !delta.HasChanges() {
+		log.V(1).Info("No delta found")
+		return nil
+	}
+
+	if delta.IsAdded() {
+		log.V(1).Info("Delta found", "create", desiredSMM.GetName())
+		if err := r.client.Create(ctx, desiredSMM); err != nil {
+			return err
+		}
+	}
+
+	if delta.IsUpdated() {
+		// Don't update a resource that we don't own.
+		if existingSMM.Labels["app.kubernetes.io/managed-by"] != "odh-model-controller" {
+			log.Error(errors.New("non suitable ServiceMeshMember resource"),
+				"There is a user-owned ServiceMeshMember resource that needs to be updated. Inference Services may not work properly.",
+				"smm.name", existingSMM.GetName(),
+				"smm.desired.smcp_namespace", desiredSMM.Spec.ControlPlaneRef.Namespace,
+				"smm.desired.smcp_name", desiredSMM.Spec.ControlPlaneRef.Name,
+				"smm.current.smcp_namespace", existingSMM.Spec.ControlPlaneRef.Namespace,
+				"smm.current.smcp_name", existingSMM.Spec.ControlPlaneRef.Name)
+			// Don't return error because it is not recoverable. It does not make sense
```

Comment:
we, we shouldn't touch resources owned by another component, instead a manual change needs to be done.